File: app/src/main/java/alejandromr92/com/movies/domain/interactor/GetPopularMoviesInteractor.java
package alejandromr92.com.movies.domain.interactor;

import alejandromr92.com.movies.domain.model.Movie;

import java.util.List;

public interface GetPopularMoviesInteractor {

    void execute(int page);

    interface Callback {
        void onPopularMoviesRetrieved(List<Movie> movieList);
        void onPopularMoviesRetrievingError(int errorCode);
    }
}

File: app/src/main/java/alejandromr92/com/movies/domain/interactor/GetPopularMoviesInteractor.kt
package alejandromr92.com.movies.domain.interactor

import alejandromr92.com.movies.domain.model.Movie
import io.reactivex.Single

interface GetPopularMoviesInteractor {
    fun execute(page: Int, onComplete: (List<Movie>) -> Unit, onError: (Throwable) -> Unit)
}